fyle_slack_app/slack/interactives/block_suggestion_handlers.py

```
@@ -0,0 +1,234 @@
+from typing import Dict, List
+
+from django.http import JsonResponse
+
+from fyle_slack_app.models.users import User
+from fyle_slack_app.fyle.expenses.views import FyleExpense
+from fyle_slack_app.libs import logger, utils
+from fyle_slack_app.slack import utils as slack_utils
+
+
+logger = logger.get_logger(__name__)
+
+
+class BlockSuggestionHandler:
+
+    _block_suggestion_handlers: Dict = {}
+
+    # Maps action_id with it's respective function
+    def _initialize_block_suggestion_handlers(self):
+        self._block_suggestion_handlers = {
+            'category_id': self.handle_category_suggestion,
+            'project_id': self.handle_project_suggestion,
+            'cost_center_id': self.handle_cost_center_suggestion,
+            'currency': self.handle_currency_suggestion,
+            'existing_report': self.handle_existing_report_suggestion
+        }
+
+
+    # Gets called when function with an action is not found
+    def _handle_invalid_block_suggestions(self, slack_payload: Dict, user_id: str, team_id: str) -> JsonResponse:
+        slack_client = slack_utils.get_slack_client(team_id)
+
+        user_dm_channel_id = slack_utils.get_slack_user_dm_channel_id(slack_client, user_id)
+        slack_client.chat_postMessage(
+            channel=user_dm_channel_id,
+            text='Looks like something went wrong :zipper_mouth_face: \n Please try again'
```

could we raise a Sentry exception failure?